src/main/java/tools/jackson/core/util/BufferRecyclerPool.java

```diff
@@ -0,0 +1,13 @@
+package tools.jackson.core.util;
+
+public class BufferRecyclerPool {
+
+    private static final ObjectPool<BufferRecycler> pool = ObjectPool.newLockFreePool(BufferRecycler::new);
```

Comment:
Does this mean that pooling would be global, across all `TokenStreamFactory` instances? If so, that would be against Jackson design in which nothing should be shared between unrelated factory instances.